dataikuapi/dss/plugin.py

```diff
@@ -116,6 +111,45 @@ def update_from_git(self, repository_url, checkout = "master", subpath=None):
         })
         return self.client.get_future(ret["jobId"])
 
+    ########################################################
+    # Plugin uninstall/delete
+    ########################################################
+
+    def list_usages(self, project_key=None):
+        """
+        Get the list of usages of the plugin.
+
+        :param str project_key: optional key of project where to look for usages. Default is None and looking in all projects.
+        :return: a :class:`DSSPluginUsages`
+        """
+        return DSSPluginUsages(
+            self.client._perform_json("GET", "/plugins/{pluginId}/actions/listUsages".format(pluginId=self.plugin_id),
+                                      params={"projectKey": project_key})
+        )
+
+    def delete(self, force=False):
+        """
+        Delete a plugin.
+
+        If not forced (default), pre-deletion checks will be run (as by :func:`prepare_delete` and the deletion will be
+        performed if and only if no usage of the plugin is detected and no error occurred during usages analysis.
+
+        :param bool force: if True, plugin will be deleted even if usages are found or errors occurred during usages
+        analysis. Default is False.
+        :return: a :class:`dataikuapi.dssfuture.DSSFuture`
+        """
+
+        params = {
+            "force": force
+        }
+        ret = self.client._perform_json("POST", "/plugins/{pluginId}/actions/delete".format(pluginId=self.plugin_id),
+                                        body=params)
+        # we check for the presence of a field of the aggregated usage statistics structure returned
+        # when the plugin exist and has usages
+        if "projectCount" in ret:
+            raise DataikuException("Plugin has usages or analysis errors. See list_usages() for more info.")
```

Comment:
This actually never happens, as we are always in one of 3 cases:
1. Delete an unused plugin: works + `ret` is a future result which does not have `projectCount`
2. Delete a used plugin with `force=True`: same
3. Delete a used plugin without `force=True`: exception is raised server-side, `_perform_json` throws

The  behavior is ok but those lines are therefore unused / a bit misleading, seems to me that we can simply remove them.

```suggestion
```